tensorflow/lite/delegates/webnn/add_test.cc

```diff
@@ -0,0 +1,935 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include <cstdint>
+#include <functional>
+#include <memory>
+#include <random>
+
+#include <gtest/gtest.h>
+#include "tensorflow/lite/delegates/webnn/binary_elementwise_tester.h"
+#include "tensorflow/lite/delegates/webnn/webnn_delegate.h"
+
+namespace tflite {
+namespace webnn {
+
+TEST(Add, 4DBy4D) {
+  TfLiteWebNNDelegateOptions delegate_options =
+      TfLiteWebNNDelegateOptionsDefault();
+  std::unique_ptr<TfLiteDelegate, decltype(&TfLiteWebNNDelegateDelete)>
+      webnn_delegate(TfLiteWebNNDelegateCreate(&delegate_options),
+                       TfLiteWebNNDelegateDelete);
+
+  std::random_device random_device;
+  auto rng = std::mt19937(random_device());
+  auto shape_rng =
+      std::bind(std::uniform_int_distribution<int32_t>(2, 5), std::ref(rng));
+  const auto batch = shape_rng();
+  const auto height = shape_rng();
+  const auto width = shape_rng();
+  const auto channels = shape_rng();
+
+  BinaryElementwiseTester()
+      .Input1Shape({batch, height, width, channels})
+      .Input2Shape({batch, height, width, channels})
+      .Test(BuiltinOperator_ADD, webnn_delegate.get());
+}
+
+TEST(Add, 4DBy4DBroadcastChannels) {
+  TfLiteWebNNDelegateOptions delegate_options =
+      TfLiteWebNNDelegateOptionsDefault();
+  std::unique_ptr<TfLiteDelegate, decltype(&TfLiteWebNNDelegateDelete)>
+      webnn_delegate(TfLiteWebNNDelegateCreate(&delegate_options),
+                       TfLiteWebNNDelegateDelete);
+
+  std::random_device random_device;
+  auto rng = std::mt19937(random_device());
+  auto shape_rng =
+      std::bind(std::uniform_int_distribution<int32_t>(2, 5), std::ref(rng));
+  const auto batch = shape_rng();
+  const auto height = shape_rng();
+  const auto width = shape_rng();
+  const auto channels = shape_rng();
+
+  BinaryElementwiseTester()
+      .Input1Shape({1, 1, 1, channels})
+      .Input2Shape({batch, height, width, channels})
+      .Test(BuiltinOperator_ADD, webnn_delegate.get());
+
+  BinaryElementwiseTester()
+      .Input1Shape({batch, height, width, channels})
+      .Input2Shape({1, 1, 1, channels})
+      .Test(BuiltinOperator_ADD, webnn_delegate.get());
+}
+
+TEST(Add, 4DBy4DBroadcastWidth) {
+  TfLiteWebNNDelegateOptions delegate_options =
+      TfLiteWebNNDelegateOptionsDefault();
+  std::unique_ptr<TfLiteDelegate, decltype(&TfLiteWebNNDelegateDelete)>
+      webnn_delegate(TfLiteWebNNDelegateCreate(&delegate_options),
+                       TfLiteWebNNDelegateDelete);
```

Comment:
Nit: indentation. Please check and fix the rest indentation issues.

```suggestion
                     TfLiteWebNNDelegateDelete);
```